docs/extend/legacy_plugins.md

```diff
@@ -15,8 +15,6 @@ keywords: "Examples, Usage, plugins, docker, documentation, user guide"
      will be rejected.
 -->
 
-# Use Docker Engine plugins
```

Comment:
We may want to consider renaming these `Docker plugins`, because IIUC, we don't refer to it as "engine" anymore, just "docker". Doesn't have to be in this PR.